pkg/compact/compact.go

```diff
@@ -913,22 +913,42 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 			return errors.Wrap(err, "build compaction groups")
 		}
 		finishedAllGroups := true
+		var wg sync.WaitGroup
+		errChan := make(chan error, len(groups))
+		groupChan := make(chan struct{}, groupCompactConcurrency)
+		defer close(groupChan)
+		for i := 0; i < groupCompactConcurrency; i++ {
```

Comment:
I am reviewing on mobile, so might miss something, but I don't get this loop. What is the purpose of it?